Generate link to most recent commit that modified file

mogelbrod opened the pull request:
This PR will generate commit links using the commit SHA that most recently modified the file, instead of simply using the most recent commit (which likely didn't touch the file in question).

knsh14 commented:
@mogelbrod Thank you for contributing!!!
How about create another function to use your awesome command.
it is intuitive that generating link with current HEAD commit's hash, but your idea is really useful.
so I want both choices.

mogelbrod committed:
Echo error when remote points to unknown host

mogelbrod committed:
Add new `GetCommitLink` command

- Update readme to list available commands
- Merge link functions into one
- Use `ref` terminology instead of `commit` to be consistent with git

mogelbrod committed:
Remove debug echo call

mogelbrod commented:
@knsh14 I turned it into another command per your request; `GetCommitLink` (dropping the `Current` part since it most likely points to an older `ref`). I also propose tweaking the error reporting a bit, does it seem reasonable?

mogelbrod committed:
Only use permalink terminology for `GetCommitLink`

knsh14 commented:
Looks great !!! 👍 
thank you!!!
